ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,336 @@
+## Contents
+* [Customizing NGINX](#customizing-nginx)
+* [Custom NGINX configuration](#custom-nginx-configuration)
+* [Custom NGINX template](#custom-nginx-template)
+* [Annotations](#annotations)
+* [Custom NGINX upstream checks](#custom-nginx-upstream-checks)
+* [Authentication](#authentication)
+* [Rewrite](#rewrite)
+* [Rate limiting](#rate-limiting)
+* [Secure backends](#secure-backends)
+* [Whitelist source range](#whitelist-source-range)
+* [Allowed parameters in configuration config map](#allowed-parameters-in-configuration-configmap)
+* [Default configuration options](#default-configuration-options)
+* [Websockets](#websockets)
+* [Optimizing TLS Time To First Byte (TTTFB)](#optimizing-tls-time-to-first-byte-tttfb)
+* [Retries in no idempotent methods](#retries-in-no-idempotent-methods)
+
+
+### Customizing nginx
+
+there are 3 ways to customize nginx
+
+1. config map: create a stand alone config map, use this if you want a different global configuration
+2. annoations: [annotate the ingress](#annotations), use this if you want a specific configuration for the site defined in the ingress rule
+3. custom template: when is required a specific setting like [open_file_cache](http://nginx.org/en/docs/http/ngx_http_core_module.html#open_file_cache), custom [log_format](http://nginx.org/en/docs/http/ngx_http_log_module.html#log_format), adjust [listen](http://nginx.org/en/docs/http/ngx_http_core_module.html#listen) options as `rcvbuf` or when is not possible to change an through the config map
+
+
+#### Custom NGINX configuration
+
+It's possible to customize the defaults in NGINX using a config map.
+
+Please check the [custom configuration](examples/custom-configuration/README.md) example
+
+#### Annotations
+
+The following annotaitons are supported:
+
+|Name                 |type|
+|---------------------------|------|
+|[ingress.kubernetes.io/rewrite-target](#rewrite)|URI|
+|[ingress.kubernetes.io/add-base-url](#rewrite)|true or false|
+|[ingress.kubernetes.io/limit-connections](#rate-limiting)|number|
+|[ingress.kubernetes.io/limit-rps](#rate-limiting)|number|
+|[ingress.kubernetes.io/auth-type](#authentication)|basic or digest|
+|[ingress.kubernetes.io/auth-secret](#authentication)|string|
+|[ingress.kubernetes.io/auth-realm](#authentication)|string|
+|[ingress.kubernetes.io/ssl-redirect](#server-side-https-enforcement-through-redirect)|true or false|
+|[ingress.kubernetes.io/upstream-max-fails](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/upstream-fail-timeout](#custom-nginx-upstream-checks)|number|
+|[ingress.kubernetes.io/secure-backends](#secure-backends)|true or false|
```

Comment:
@mgoodness the name will change to avoid specific nginx or haproxy terminologies
(suggested by @bprashanth)